zwave_js_server/model/node/__init__.py

```diff
@@ -464,10 +466,23 @@ async def async_send_command(
             kwargs["require_schema"] = require_schema
 
         if wait_for_result or (
-            wait_for_result is None and self.status != NodeStatus.ASLEEP
+            wait_for_result is None
+            and self.status not in (NodeStatus.ASLEEP, NodeStatus.DEAD)
         ):
-            result = await self.client.async_send_command(message, **kwargs)
-            return result
+            result_task = asyncio.create_task(
+                self.client.async_send_command(message, **kwargs)
+            )
+            await asyncio.wait(
```

Comment:
Should we ignore the event if `wait_for_result` is set to True?

Reply:
no I think that's one of the core issues. We can't afford to wait for a response when a device is asleep/dead, otherwise it will block forever. So `wait_for_result` shouldn't matter for most commands that are sent

Reply:
Then the usefulness of this parameter isn't clear to me. Why should we be able to start waiting for a command to a node that is sleeping or is dead if we won't finish waiting for the command if the node fell asleep or died?

Reply:
yeah that's a fair point and I'm not sure how to address it. There are commands that need a response, but there are also commands sent when the node status is effectively incorrect. So what do we do when it corrects itself? maybe we should only cancel if `wait_for_result` is set to `None`?

Reply:
OK I guess we should not follow this path when `wait_for_result` is `True`. Worst case scenario, nothing changes for certain calls. I looked through the calls that use it and I don't see anything that would block your typical services, those are almost always going to be set_value calls (the one exception I can find is for retrieving firmware information but I doubt anyone is using that service in an automation, or if they are, they probably shouldn't)